lenovo/thinkpad/t14s/amd/gen1/default.nix

```diff
@@ -0,0 +1,8 @@
+
+{ config, lib, pkgs, ... }:
+
+{
+  imports = [
```

Comment:
Maybe we should add this once, we have gen2 and it becomes necessary that we need different configuration?

Reply:
Is there already a second generation? What if Lenovo never releases one?

Reply:
What I mean to say, I would only introduce `lenovo/thinkpad/t14s/amd/default.nix` for now and add `gen1` later if newer models come with different needs.

Reply:
I was thinking of the case where you have <nixos-hardware/lenovo/thinkpad/t14s/amd> `imports` and then if we have to do the split into `gen1` and `gen2`, the import path breaks, because the specific parts for `gen1` go into `./gen1`. Or is there a way to later notify the user that they maybe want to include the more specific file instead?

Reply:
I think we could add a warning in this case, by defining some options inside the `gen1` and `gen2` modules.

Reply:
If `/etc/nixos/configuration.nix` contained `import = [ <nixos-hardware/lenovo/thinkpad/t14s/amd> ];`, then `./gen1` and `./gen2/` would not be evaluated and the warning will never be shown.

Reply:
Also Lenovo lists the laptop as `Gen1` everywhere, so that this heavily implies that there may be a Gen2 at some point along with a new T13 and T15.